worker/src/docker/taskbridge/bridge.rs

```diff
@@ -114,23 +150,83 @@ impl TaskBridge {
                             while current_pos < trimmed.len() {
                                 // Try to find a complete JSON object
                                 if let Some(json_str) = extract_next_json(&trimmed[current_pos..]) {
-                                    match serde_json::from_str::<MetricInput>(json_str) {
-                                        Ok(input) => {
-                                            println!("Received metric: {:?}", input);
-                                            let _ = store
-                                                .update_metric(
-                                                    input.task_id,
-                                                    input.label,
-                                                    input.value,
+                                    debug!("Received metric: {:?}", json_str);
+                                    if json_str.contains("file_name") {
+                                        let storage_path = match &storage_path_clone {
+                                            Some(path) => path,
+                                            None => {
+                                                println!(
+                                                    "Storage path is not set - cannot upload file."
+                                                );
+                                                continue;
+                                            }
+                                        };
+
+                                        if let Ok(file_info) =
+                                            serde_json::from_str::<serde_json::Value>(json_str)
+                                        {
+                                            if let Some(file_name) = file_info["value"].as_str() {
+                                                let task_id =
+                                                    file_info["task_id"].as_str().unwrap();
+
+                                                if let Err(e) = file_handler::handle_file_upload(
```

Comment:
there's no rate limiting here right? in the permissionless scenario, someone could flood this, probably worth setting some parameter that makes sense re how often a node/IP should be able to call this function - maybe also set a limit to the file sizes?